src/Rule/Length.php

```diff
@@ -62,8 +71,12 @@ public function __construct($length)
      */
     public function validate($value)
     {
-        $actualLength = strlen($value);
-
+        if (is_null($this->encoding) || !function_exists('mb_strlen')) {
+            $actualLength = strlen($value);
+        } else {
+            $actualLength = mb_strlen($value, $this->encoding);
```

Comment:
Why not to use just `$actualLength = function_exists('mb_strlen') ? mb_strlen($value) : strlen($value)`?

I want to just use `->length(10)` with default encoding, I don't want to pass `'utf8'` everywhere.

Reply:
You are right. I just tried to be **backwards compatible**. Maybe someone uses the rule and knows about that non-multibyte behaviour. Then this person maybe decided to count the number of special chars like äöüéàè and simple removes that number from the length (because in utf8, these would get treated as two chars respecively).

But of course, it would be better to handle this as default.

Reply:
@adriansuter I agree that your implementation is backward compatible.

This is an ambiguous question. I think that counting bytes (not characters) is a bug, but other may think vice versa because the documentation is not precise about it. Hope that this question will be clarified in the next major release.

Reply:
Me too. I think PHP should change the default behaviour of `strlen()`. The expected result for most people I suppose, is the number of characters. In PHP (in case PHP is used in web development) one rarely has to count the byte length of a string. And if so, there is always `unpack()`.

We will see, if PHP would implement that (I doubt it :-)).